.github/workflows/release.yml

```diff
@@ -0,0 +1,50 @@
+name: Build and release
+
+on:
+  push:
+    branches: [ v2.x.x ]
+  workflow_dispatch:
+    inputs:
+      snapshot_version:
+        description: 'The snapshot version that is going to be published'
+        required: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{ secrets.ZOWE_ROBOT_TOKEN }}
+      - uses: ./.github/actions/native-build
+        with:
+          host: ${{ secrets.SSH_MARIST_RACF_HOST }}
+          username: ${{ secrets.SSH_MARIST_USERNAME }}
+          passwd: ${{ secrets.SSH_MARIST_RACF_PASSWORD }}
+
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Use Node LTS
+        uses: actions/setup-node@v3
+        with:
+          node-version: 'lts/*'
+      - name: install zowe/cli
+        run: npm install -g @zowe/cli@6.32.2
+      - name: install zowe-api-dev
+        run: npm -g install @zowedev/zowe-api-dev
+
+      - name: Release or snapshot
+        run: |
+          if [ ${{github.event.inputs.snapshot_version }} ];
+          ./gradlew publish -x test -Pzowe.deploy.username=$ARTIFACTORY_USERNAME -Pzowe.deploy.password=$ARTIFACTORY_PASSWORD -Partifactory_user=$ARTIFACTORY_USERNAME -Partifactory_password=$ARTIFACTORY_PASSWORD
+          else
+          ./gradlew release -x test -Prelease.useAutomaticVersion=true -Pzowe.deploy.username=$ARTIFACTORY_USERNAME -Pzowe.deploy.password=$ARTIFACTORY_PASSWORD -Partifactory_user=$ARTIFACTORY_USERNAME -Partifactory_password=$ARTIFACTORY_PASSWORD
```

Comment:
how is this automatic version going to work?

Reply:
it was not done properly, I have split it into release and publish, release will be done by initiating the workflow manualy